### PR TITLE
Load SSCT Generation Counter Upon DR Promotion [OSS]

### PR DESCRIPTION
**Pull request by HridoyRoy**
This PR is a port of https://github.com/hashicorp/vault-enterprise/pull/3170/files.

Description (from ent PR): 
This PR fixes the regression with SSCT. 

The issue is that the SSCT Generation counter was not being loaded from storage, so when it was written to storage during DR promotion it would always be 1. The fix is to load the value from storage during DR promotion. DRs never use the generation counter, but on DR clusters the value in memory will always be 1 less than the value in storage. However, the moment the DR gets updated to a primary, the value from storage will be incremented and written back.

Will open the OSS port after this PR is approved, with a changelog in OSS.

The PR will need to be backported to 1.10.

### CODE REVIEW

File: changelog/16956.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+core: Load SSCT Token Generation counter from storage when upgrading a DR to a primary
```
Comment (ncabatoff) [inline changelog/16956.txt:2]: This describes the change but isn't meaningful to most changelog readers.  Can you instead describe the change from a user perspective?

Comment (HridoyRoy) [reply changelog/16956.txt]: Done!

**Review by ncabatoff — APPROVED** [review]

**Review by raskchanky — APPROVED** [review]

**Comment by mladlow** [comment]
The associated enterprise changes for this fix didn't merge alongside the OSS changes. The enterprise changes will be released in 1.10.10. The whole change was released in 1.11.4.

**Comment by mladlow** [comment]
Upon further review, it looks like the enterprise changes are tests. Those have merged into 1.10.x now on the enterprise side, so it's good that this is complete, but I think the functionality was fully released in 1.10.7.

**Pull request merged**